libraries/aws/provisioning/src/aws_iot_provisioning_api.c

```diff
@@ -617,10 +617,10 @@ AwsIotProvisioningError_t AwsIotProvisioning_CreateCertificateFromCsr( IotMqttCo
         /* Serialization of request payload occurs in a 2-step process, one for calculation of buffer size, and then,
```

Comment:
I'd add to this comment that the function returns the size even when the buffer address is `NULL`.

Reply:
Created a separate function for doing the calculation of serialization payload size for better readability and single-responsibility of the serializer function!